### PR TITLE
Optimized making requests when selecting users' apps

### PR DESCRIPTION
**Pull request by scorpion35**
By calling `Application.all` once, the GET request is only executed once per user, when `selected_apps` is called. Thanks @janpio for the tip!

Branch: "master" | Commit Type: "Bug fix" | Changes tested: "yes"

---

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
**Context:**
- I am trying to get all users in a account and fetch the app names each user has access to. 

**Problem:**
- If a user has access to 10 apps (as an example), `10` GET requests were sent to Apple.
- And if there are 100 users/account, `100 * 10 = 1000` GET requests would be sent to Apple.

**If it fixes an open issue, please link to the issue here.**
- https://github.com/fastlane/fastlane/issues/14070
- This doesn't fix the issue completely, but optimizes **the number of GET requests to only one per user** (which is a huge win I think).

### Description
- Get all `Applications` before the loop. `all_apps = Application.all`
- Run `select` on the local `all_apps` variable.

**Please describe in detail how you tested your changes.**
- I have about over 600 users in all of our accounts combined. When I ran my script for all users, I have only seen ~600 GET requests.
- I have also noticed it's very fast now, even if the user has access to over 250 apps.

### CODE REVIEW

**Comment by scorpion35** [comment]
Pushed this from my work computer, probably that's why the CLA check is getting confused? I can confirm only I have pushed this fix!

**Comment by scorpion35** [comment]
Do you know how I can sign then CLA? Tried looking for the link, but wasn't able to find it.

File: spaceship/lib/spaceship/tunes/member.rb
Comment (joshdholtz) [inline spaceship/lib/spaceship/tunes/member.rb]:
```suggestion
          parsed_apps << all_apps.select { |app| app.apple_id == app_id }
```
You probably don't need to check for `nil` here ^ nil will never equal `app_id` so this will clean it up a bit

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One small nitpick on a nil check otherwise this looks 👍

**Comment by janpio** [comment]
(Argh, I should actually read what @googlebot writes: As the Github account that was used to create the PR is not the same as the one that was used to push, it will not check for the CLA for either one. @joshdholtz will have to decide if we can merge this in this state, or if we need a force push with your GitHub account and then probably a new PR to fix this)

**Comment by scorpion35** [comment]
haha no worries. I will wait for the word!

**Comment by janpio** [comment]
> Also, do you have any update on my PR, please? I am currently referencing my script to local fastlane changes, would have to move it to Jenkins build farm soon. Would appreciate if it's merged soon.

I can't influence when or if we get word on the CLA problem. Quickest would probably be to force push to the branch so it is committed/pushed by your Github account and then close this one and open a new PR where the bot will be able to connect you to the CLA properly - then I can just merge your PR.

**Comment by scorpion35** [comment]
@janpio I have tried several things, but it seems Github is always coming back to these changes. Can this PR be merged with an approver this time, please? I will submit from my personal computer from next time.

@joshdholtz 🙏 Sorry for mentioning, just hoping for an approval so the PR can be merged :)

**Comment by scorpion35** [comment]
Alternatively if you want to submit from your account, I am ok with that too :)

**Comment by joshdholtz** [comment]
@scorpion35 Looks like Google says its okay 👌 ` CLAs are signed, but unable to verify author consent ` and you consent so you are 👍 Doing a one last quick review and test then we will merge

**Comment by joshdholtz** [comment]
@scorpion35 It actually looks like there are no changes in this PR anymore 🤔 

screen shot 2019-01-20 at 1 15 49 pm

**Comment by janpio** [comment]
(I am sure @joshdholtz will take care of this now that he is aware. 

I looked at your 2 other PRs: 

The initial commits were not changed as I could see. If I were in your situation, I would have reset my branch to the first commit _before_ you committed anything with the wrong account, then committed the changes (that were kept in my working copy) again with the correct user - and finally force (!) pushed to my branch to get rid of the old commit and replace them with the new ones. That should automatically have updated this PR, but to get @googlebot to look at it again you still would have needed to create a new PR. 

But yep, that is definitely one of those git workflows that are not very common and stuff can go wrong. With the git CLI I would have no idea how to do it to be honest, but I work with Sourcetree, a git UI, long enough to be able to do this consistently now. Thank god ;)

Thanks for trying!)

**Pull request closed without merging**
